docs/README-Extending.md

```diff
@@ -0,0 +1,136 @@
+         */
+        getProfile(primaryNode: IZoweTreeNode): IProfileLoaded;
+        
+        /**
+         * Used by other VS Code Extensions to access an alternative
+         * profile types that can be employed in conjunction with the primary
+         * profile to provide alternative support.
+         *
+         * @param primaryNode represents the Tree item that is being used
+         * @return The requested profile
+         */
+        getLinkedProfile(primaryNode: IZoweTreeNode, type: string): Promise<IProfileLoaded>;
+```
+
+## The relationship with CLI profiles
+
+Zowe Explorer uses the Zowe CLI profiles that have been created by the user and accesses information via a primary profile such as z/OSMF
+
+!["Primary Profile"](./images/ZE-basic.gif?raw=true "Primary Profile")
+
+This diagram shows that information presented in the Zowe Explorer has been retrieved from a z/OS server via an instance of a z/OSMF profile. Because
+Zowe Explorer embeds the Zowe CLI node modules it uses this profile in conjunction with the API layer of the CLI plugin to create the
+displayed content and provide functionality such as menus and other options.
+
+There are of course other sources of information that could be presented that is not available via the primary profile. For example specific vendor tools associated with
+Jobs or functions that represent advanced file management. It is intented to we "open up" Zowe Explorer and allow alternative routes to access these supplementary sources. 
+The basis of an extender's extension therefore is to use an alternative secondary information source based upon a Zowe CLI plugin that will allow. Zowe explorer will manage the
+relationship between a primary and secondary profiles and the Zowe Extender API provides methods to access both the primary profile and a secondary profile based on the type.   
+
+The result is that from within the scope of the extender package it's possible to access function from either source.
+
+!["Secondary Profile"](./images/ZE-extend.gif?raw=true "Secondary Profile")
+
+The information can be presented via specific menu options or potentially via callback functions for example during an open request. These access methods API's are to be developed 
+based upon extender requirements which can be discussed via our Zowe slack channel or by raising a [git issue](https://github.com/zowe/vscode-extension-for-zowe/issues)  
+
+
+## Associating Zowe CLI Profiles
+
+Zowe Explorer is built on Zowe CLI which is the preferred method for accessing additional REST services. To allow extenders to bring their own 
+Zowe CLI plugins there is a simple mechanism by which Zowe Explorers can associate profiles. The 'Primary' profile which is powering Zowe Explorer
+can be linked with multiple 'Secondary' types. But at this time only a single instance of each type is allowed. 
+from within their own extender package these profiles can be requested and so making access to the relevant REST API available.
```

Comment:
The "But at this time only a single instance of each type is allowed." and "from within their own extender package..." should be linked, correct?

Reply:
Reworded "The 'Primary' profile which is powering Zowe Explorer
can be linked with multiple 'Secondary' types although only a single instance of each type is allowed. "